Source/MLX/MLXArray+Bytes.swift

```diff
@@ -0,0 +1,314 @@
+// Copyright © 2024 Apple Inc.
+
+import Cmlx
+import Foundation
+import Metal
+
+// MARK: - Backing / Bytes
+
+extension MLXArray {
+
+    /// Return the dimension where the storage is contiguous.
+    ///
+    /// If this returns 0 then the whole storage is contiguous.  If it returns ndmin + 1 then none of it is contiguous.
+    func contiguousToDimension() -> Int {
+        let shape = self.shape
+        let strides = self.internalStrides
+
+        var expectedStride = 1
+
+        for (dimension, (shape, stride)) in zip(shape, strides).enumerated().reversed() {
+            // as long as the actual strides match the expected (contiguous) strides
+            // the backing is contiguous in these dimensions
+            if stride != expectedStride {
+                return dimension + 1
+            }
+            expectedStride *= shape
+        }
+
+        return 0
+    }
+
+    /// Return the physical size of the backing (assuming it is evaluated) in elements.  This should
+    /// only be used when accessing the backing directly, e.g. via `mlx_array_data_uint8()`
+    var physicalSize: Int {
+        // nbytes is the logical size of the input, not the physical size
+        return zip(self.shape, self.internalStrides)
+            .map { Swift.abs($0.0 * $0.1) }
+            .max()
+            ?? self.size
+    }
+
+    func copy(from: UnsafeRawBufferPointer, toContiguous output: UnsafeMutableRawBufferPointer) {
```

Comment:
This is the same as before with the output label now indicating that it is contiguous storage.